ProjectLighthouse.Servers.API/Controllers/UserEndpoints.cs

```diff
@@ -50,6 +50,32 @@ public async Task<IActionResult> GetUser(string username)
         return this.Ok(user);
     }
 
+    /// <summary>
+    /// Searches for the user based on the query
+    /// </summary>
+    /// <param name="query">The search query</param>
+    /// <returns>A list of users</returns>
+    /// <response code="200">The list of users, if any were found</response>
+    /// <response code="404">No users matched the query</response>
+    [HttpGet("user_search")]
+    [ProducesResponseType(typeof(User), StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<IActionResult> SearchUsers(string query)
+    {
+        List<User> users = await this.database.Users
+            .Where(u => u.PermissionLevel != PermissionLevel.Banned && u.Username.Contains(query))
```

Comment:
How does the .Contains() here play with uppercase/lowercase, is this a concern or is it expected that you should capitalize the username properly?

Reply:
capitalisation seems to be ignored

Reply:
Should be fine then